src/main/java/AdminTools/AdminTools.java

```diff
@@ -4,40 +4,36 @@
 import AdminTools.ClearChat.ClearChatCommand;
 import AdminTools.Fly.FlyCommand;
 import AdminTools.Freeze.FreezeCommand;
+import AdminTools.Punishment.TempBan;
 import AdminTools.Vanish.VanishCommand;
+import org.bukkit.ChatColor;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandExecutor;
+import org.bukkit.command.CommandSender;
 import org.bukkit.entity.Player;
 import org.bukkit.event.EventHandler;
 import org.bukkit.event.Listener;
 import org.bukkit.event.player.AsyncPlayerChatEvent;
 import org.bukkit.plugin.java.JavaPlugin;
 
-
 public final class AdminTools extends JavaPlugin implements Listener {
 
     private ChatLockCommand chatLockCommand;
     public static String messagePrefix;
-    private FlyCommand flyCommand;
-    private VanishCommand vanishCommand;
 
     @Override
     public void onEnable() {
         messagePrefix = "§7[§cAdminTools§7] ";
         // Initialize and register the ChatLockCommand
         chatLockCommand = new ChatLockCommand(this);
-        vanishCommand = new VanishCommand(); // Add this line
-        flyCommand = new FlyCommand(); // Add this line
-        getCommand("admin").setExecutor(new AdminCommand(vanishCommand, flyCommand));
         getCommand("clearchat").setExecutor(new ClearChatCommand());
-        getCommand("vanish").setExecutor(vanishCommand); // Update this line
-        getCommand("fly").setExecutor(flyCommand); // Update this line
+        getCommand("fly").setExecutor(new FlyCommand());
+        getCommand("vanish").setExecutor(new VanishCommand());
         getCommand("freeze").setExecutor(new FreezeCommand(this));
+        String commandName = "tempban";
+        getCommand(commandName).setExecutor(new TempBan(this, commandName, null));
```

Comment:
Why have you created a string `commandName` that holds the command name? Dont mix different code styles please. 

```suggestion
        // TODO Remember to change the constructor of TempBan, so it only needs Plugin
        getCommand(commandName).setExecutor(new TempBan(this, "tempban", null));
```